### PR TITLE
[beta 1.69] Backport #11881 Add delays to network retries.

### PR DESCRIPTION
**Pull request by ehuss**
This is a backport of #11881 to beta. The intent is to help mitigate any potential issues that might crop up with turning sparse index on by default.

This also includes some other changes to get CI passing:
* #11916 — Disable test_profile test on windows-gnu
* Ignored some tests due to a change in clap output

### CODE REVIEW

**Comment by ehuss** [comment]
@rust-lang/cargo This is a proposal for a beta backport. I'm not entirely confident this is a good idea, since networking code is generally delicate, and this has risk that it will introduce some regressions. It is also a fairly large amount of code. However, I worry about potential caching issues with the sparse rollout (though there has only been one report of an issue). The change should land in nightly tomorrow, so at least it will git a little bit of testing there. 

Let me know whether or not you think we should move forward with this backport.

File: src/cargo/sources/git/oxide.rs [hunk 1]
```diff
@@ -0,0 +1,355 @@
+//! This module contains all code sporting `gitoxide` for operations on `git` repositories and it mirrors
```
[the remark this answers is not in the document]
Comment (ehuss) [reply src/cargo/sources/git/oxide.rs]: I'm not sure what happened, I somehow mistakenly added this file. It has been removed.

**Review by epage — APPROVED** [review]
Ok with merging this if we agree to the backport

**Review by weihanglo — APPROVED** [review]
Seems good to me, thanks!

If we want to backport this, it should be done as early as possible.

**Comment by ehuss** [comment]
Let's discuss the beta backports at today's meeting.

**Pull request closed without merging**